packages/omgidl-serialization/src/DeserializationInfoCache.ts

```diff
@@ -269,6 +382,43 @@ export const PRIMITIVE_ARRAY_DESERIALIZERS = new Map<string, ArrayDeserializer>(
   ["string", readStringArray],
 ]);
 
+export const PRIMITIVE_DEFAULT_VALUE_GETTERS = new Map<string, () => unknown>([
+  ["bool", () => DEFAULT_BOOLEAN_VALUE],
+  ["int8", () => DEFAULT_BYTE_VALUE],
+  ["uint8", () => DEFAULT_BYTE_VALUE],
+  ["int16", () => DEFAULT_NUMERICAL_VALUE],
+  ["uint16", () => DEFAULT_NUMERICAL_VALUE],
+  ["int32", () => DEFAULT_NUMERICAL_VALUE],
+  ["uint32", () => DEFAULT_NUMERICAL_VALUE],
+  ["int64", () => DEFAULT_NUMERICAL_VALUE],
+  ["uint64", () => DEFAULT_NUMERICAL_VALUE],
+  ["float32", () => DEFAULT_NUMERICAL_VALUE],
+  ["float64", () => DEFAULT_NUMERICAL_VALUE],
+  ["string", () => DEFAULT_STRING_VALUE],
+]);
+
+export function makeNestedArray(
+  getValue: () => unknown,
```

Comment:
I think you can just pass `value: unknown` here instead of a getter function?

Reply:
I use getValue instead of `value` here because I want to prevent assigning all of the values to the same reference if the value is not a primitive.